tools/ci-build/runtime-versioner/src/audit.rs

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::{
+    command::CommandExt,
+    index::CratesIndex,
+    repo::Repo,
+    tag::{previous_release_tag, release_tags},
+    util::utf8_path_buf,
+    Audit,
+};
+use anyhow::{anyhow, bail, Context, Result};
+use camino::{Utf8Path, Utf8PathBuf};
+use smithy_rs_tool_common::release_tag::ReleaseTag;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    fs,
+    process::Command,
+};
+
+pub fn audit(args: Audit) -> Result<()> {
+    let repo = Repo::new(args.smithy_rs_path.as_deref())?;
+    if !args.no_fetch {
+        // Make sure we have the latest release tags
+        fetch_smithy_rs_tags(&repo)?;
+    }
+
+    let release_tags = release_tags(&repo)?;
+    let previous_release_tag =
+        previous_release_tag(&repo, &release_tags, args.previous_release_tag.as_deref())?;
+    if release_tags.first() != Some(&previous_release_tag) {
+        tracing::warn!("there are newer releases since '{previous_release_tag}'");
+    }
+
+    let next_crates = discover_runtime_crates(&repo.root).context("next")?;
+    let previous_crates = resolve_previous_crates(&repo, previous_release_tag.as_str())?;
+
```

Comment:
if you want to run this locally you could also stash and unstash

Reply:
I posted an update that checks out the previous revision to a temp directory so that it can work locally.